authz/grpc_audit_end2end_test.go

```diff
@@ -0,0 +1,344 @@
+package authz
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io"
+	"net"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/internal/grpctest"
+	testgrpc "google.golang.org/grpc/interop/grpc_testing"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/testdata"
+
+	_ "google.golang.org/grpc/authz/audit/stdout"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+type testServer struct {
+	testgrpc.UnimplementedTestServiceServer
+}
+
+func (s *testServer) UnaryCall(ctx context.Context, req *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	return &testpb.SimpleResponse{}, nil
+}
+
+type statAuditLogger struct {
+	AuthzDescisionStat map[bool]int      //Map to hold the counts of authorization decisions
+	EventContent       map[string]string //Map to hold event fields in key:value fashion
+	SpiffeIds          []string          //Slice to hold collected SPIFFE IDs
+}
+
+func (s *statAuditLogger) Log(event *audit.Event) {
+	if event.Authorized {
+		s.AuthzDescisionStat[true]++
```

Comment:
nit: Could do `s.AuthzDecisionStat[event.Authorized]++`. Also there is a typo for `Decision` here.